vaadin-messages-flow-parent/vaadin-messages-flow/src/test/java/com/vaadin/flow/component/messages/tests/MessageListTest.java

```diff
@@ -67,4 +74,31 @@ public void setItems_nullCollection_throws() {
     public void setItems_containsNullItem_throws() {
         messageList.setItems(item1, null, item2);
     }
+
+    @Test
+    public void localeChanged_schedulesItemsUpdate() {
+        SpyMessageList spyMessageList = new SpyMessageList();
+        UI ui = new UI();
+        ui.getInternals().setSession(mock(VaadinSession.class));
+        LocaleChangeEvent localeChangeEvent = new LocaleChangeEvent(ui, ui.getLocale());
+        spyMessageList.localeChange(localeChangeEvent);
+        ui.add(spyMessageList);
+        ui.getInternals().getStateTree().runExecutionsBeforeClientResponse();
+        Element spyElement = spyMessageList.getElement();
+        verify(spyElement).executeJs(eq("window.Vaadin.Flow.messageListConnector.setItems(this, $0, $1)"), any(JsonArray.class), eq(ui.getLocale().toLanguageTag()));
```

Comment:
I'm not sure if this is better than integration tests after all, because this depends on the implementation details (calling the JS function), and it doesn't guarantee that such a JS function even exists (this test would pass if we rename `setItems` to `foo`). :thinking:

Reply:
Yeah, I'm now setting/changing the locale directly on the UI in integrations tests.